posthog/temporal/data_imports/sources/linkedin_ads/service/service.py

```diff
@@ -0,0 +1,318 @@
+
+        Args:
+            client: LinkedIn Ads client instance
+
+        Returns:
+            Dictionary mapping resource names to (method, pivot) tuples
+        """
+        return {
+            LinkedinAdsResource.CampaignStats: (client.get_analytics, "CAMPAIGN"),
+            LinkedinAdsResource.CampaignGroupStats: (client.get_analytics, "CAMPAIGN_GROUP"),
+            LinkedinAdsResource.Campaigns: (client.get_campaigns, None),
+            LinkedinAdsResource.CampaignGroups: (client.get_campaign_groups, None),
+            LinkedinAdsResource.Accounts: (client.get_accounts, None),
+        }
+
+    def _prepare_analytics_dates(
+        self,
+        should_use_incremental_field: bool,
+        incremental_field: Optional[str],
+        incremental_field_type: Optional[IncrementalFieldType],
+        db_incremental_field_last_value: Any,
+        date_start: Optional[str],
+        date_end: Optional[str],
+        sync_frequency_interval: Optional[dt.timedelta],
+    ) -> tuple[Optional[str], Optional[str]]:
+        """Prepare date range for analytics requests.
+
+        Args:
+            should_use_incremental_field: Whether to use incremental sync
+            incremental_field: Field name for incremental sync
+            incremental_field_type: Field type for incremental sync
+            db_incremental_field_last_value: Last value for incremental sync
+            date_start: Start date for analytics data
+            date_end: End date for analytics data
+            sync_frequency_interval: Sync frequency interval
+
+        Returns:
+            Tuple of (date_start, date_end) strings
+        """
+        if should_use_incremental_field and incremental_field and incremental_field_type:
+            # Determine last value using incremental_field_type
+            if db_incremental_field_last_value is None:
+                last_value = (dt.datetime.now() - dt.timedelta(days=30)).strftime("%Y-%m-%d")
+            else:
+                last_value = db_incremental_field_last_value
+
+            # For analytics (date-based), use incremental value as start date
+            if (
+                incremental_field_type == IncrementalFieldType.Date
+                and incremental_field == "dateRange.start"
+                and not date_start
+            ):
+                date_start = self.date_handler.calculate_incremental_date_range(last_value, sync_frequency_interval)
+
+        return date_start, date_end
+
+    def _create_source_response(self, resource_name: str, data: list[dict[str, Any]]) -> SourceResponse:
+        """Create SourceResponse object from fetched data.
+
+        Args:
+            resource_name: Name of the resource
+            data: Raw data from the API
+
+        Returns:
+            SourceResponse object for the pipeline
+        """
+        # Flatten the data structure
+        flattened_data = [flatten_data_item(item, resource_name) for item in data]
```

Comment:
This is loading _all_ the data from the API before yielding to the pipeline - this is bad and can cause memory issues in the pods. See https://github.com/PostHog/posthog/blob/master/posthog/temporal/data_imports/sources/README.md#source-logic - but the point of the iterator function of `items` is to yield either single items or pages of items to the pipeline asap